py/usageaccountant/datadog_fetcher.py

```diff
@@ -0,0 +1,309 @@
+import argparse
+import json
+import logging
+import os
+import urllib.parse
+from typing import Any, Mapping, Optional, Sequence, TextIO, Tuple, TypedDict
+from urllib.request import Request, urlopen
+
+from usageaccountant.accumulator import UsageAccumulator, UsageUnit
+
+headers = {
+    "DD-APPLICATION-KEY": os.environ.get("DATADOG_APP_KEY", ""),
+    "DD-API-KEY": os.environ.get("DATADOG_API_KEY", ""),
+}
+
+logger = logging.getLogger("fetcher")
+
+# TODO remove handlers before prod
+logging.basicConfig(
+    level=logging.INFO,
+    format="%(asctime)s [%(levelname)s] %(message)s",
+    handlers=[logging.FileHandler("debug.log"), logging.StreamHandler()],
+)
+
+
+class DatadogResponseUnit(TypedDict):
+    plural: str
+    name: str
+
+
+class DatadogResponseSeries(TypedDict):
+    unit: Sequence[DatadogResponseUnit]
+    pointlist: Sequence[Sequence[float]]
+    scope: str
+
+
+class DatadogResponse(TypedDict):
+    error: Optional[str]
+    query: str
+    from_date: int
+    to_date: int
+    status: str
+    series: Sequence[DatadogResponseSeries]
+
+
+class InvalidResponseError(Exception):
+    def __init__(self, message: str, response: Mapping[Any, Any]) -> None:
+        self.query = response.get("query")
+        self.from_date = response.get("from_date")
+        self.to_date = response.get("to_date")
+        self.message = message
+
+    def str(self) -> str:
+        return f""" {self.message}
+                    Query: {self.query},
+                    from_date: {self.from_date},
+                    to_date: {self.to_date}."""
```

Comment:
This is adding 18 spaces before `Query:` when printing. Is that intentional ? 
```
Type "help", "copyright", "credits" or "license" for more information.
>>> a = f""" bla
...                     Query: bla,
...                     from_date: bla,
...                     to_date: bla."""
>>> a
' bla\n                    Query: bla,\n                    from_date: bla,\n                    to_date: bla.'
>>> print(a)
 bla
                    Query: bla,
                    from_date: bla,
                    to_date: bla.

```

Reply:
At first I thought it'd look cleaner, but now I don't think so. Changed it to be one line.